File: kata_11th_feb/pythonTasks/myqueue.py
class MyQueue:
    def __init__(self):
        self.number = 0
        self.elements = []

    def is_empty(self):
        return self.size == 0

    def is_not_empty(self):
        return self.size != 0

    def enqueue(self, element):
        self.elements.append(element)

    def size(self):
        return len(self.elements)

    def dequeue(self):
        if not self.is_empty():
            value = self.elements[0]
            self.elements.pop(0)
            return value
        raise IndexError("cannot dequeue an empty queue")
        

    def offer(self, element):
        self.elements[self.size] = element
        self.size += 1

    def peek(self):
        if not self.is_empty():
            return self.elements[0]
        raise IndexError("Queue is empty")

    def poll(self):
        if not self.is_empty():
            value = self.elements[0]
            self.elements.pop(0)
            return value
        return None

File: kata_11th_feb/MyQueue.js
class MyQueue {
    constructor() {
        this.isEmpty = true;
        this.size = 0;
        this.number = 0;
        this.elements = new Array(5);
    }

    isEmpty() {
        this.isEmpty = true;
        return this.isEmpty;
    }

    isNotEmpty() {
        this.isEmpty = true;
        return this.isEmpty;
    }

    add(element) {
        this.elements[this.size] = element;
        this.size++;
    }

    size() {
        return this.size;
    }

    remove(element) {
        const element1 = this.elements[this.size - (this.size - 1)];
        this.size--;
        this.number++;
    }

    offer(element) {
        this.elements[this.size] = element;
        this.size++;
    }

    peek() {
        return this.elements[this.size - 1];
    }

    poll() {
        return this.elements[this.size - 1];
    }
}
